### DISCUSSION TITLE
An example calling the Swagger API?

### DISCUSSION THREAD

**Discussion by ADefWebserver** [oqtane/oqtane.framework#1078] [Q&A] [answered]
Is there an example calling the Swagger API?
For example at: http://localhost:44357/swagger/index.html

For example, if the Blog Module is installed, it exposes this:

![image](https://user-images.githubusercontent.com/1857799/105652810-11ec1f00-5e6f-11eb-979b-886318a2c2e8.png)

However, I can not come up with the proper values to pass to get any response other than an error.
Also, How would I call a method that required authentication?

**Comment by sbwalker** [comment]
Swagger has a lot of options to enable functionality in the auto-generated API user interface. I think we need a Swagger expert to help with the configuration.

**Reply by ADefWebserver** [reply]
@sbwalker - Do you think it is a Swagger issue, or, an issue with the underlying API methods? 

Meaning, I don't think the required [httpContext ](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/http-context?view=aspnetcore-5.0#blazor-and-shared-state)is being passed (or I don't know how to pass it)

The API methods clearly work fine when called by the Blazor client (either the Blazor Server Side or the Blazor WebAssembly client). However, when _called from outside of Oqtane_, all the API methods seem to fail :(

**Comment by sbwalker** [comment, marked as answer]
@ADefWebserver due to the multi-tenant aspect of Oqtane, calls to the back-end API need to identify the tenant/site. This is currently supported by passing the AliasId as the first element in the Url path ( in your screen shot above it says "site" but the Oqtane core API uses /{alias}/api/entity ). Based on the AliasId the system is able to lookup the Tenant and Site in the database. In addition to multi-tenancy there is also authentication and authorization requirements. Currently Oqtane only supports cookie auth - so if you want to call the API you need to have a valid auth cookie. In addition, most API methods require authorization. Authorization is based on the SiteId and it retrieves permissions from the Tenant database based on the permission type and entityid.

**Reply by ADefWebserver** [reply]
@sbwalker - Thank you for the direction. I will return to this in the next few weeks (I am deep into the Oqtane Survey module right now). 

FYI - This question is because I am trying to add [LiveWriter support to the Blog module](https://github.com/ADefWebserver/oqtane.blogs/tree/windows-live-writer-support).

Thank You
